Source/CarthageKit/Errors.swift

```diff
@@ -19,7 +19,7 @@ public enum CarthageError: ErrorType, Equatable {
 	case MissingBuildSetting(String)
 
 	/// Incompatible version specifiers were given for a dependency.
-	case IncompatibleRequirements(ProjectIdentifier, VersionSpecifier, VersionSpecifier)
+	case IncompatibleRequirements(ProjectIdentifier, (VersionSpecifier, ProjectIdentifier?), (VersionSpecifier, ProjectIdentifier?))
```

Comment:
Maybe it would be worth creating an inner struct to represent this tuple? Or at least a typealias with names elements.